### PR TITLE
System.Transactions, enable strict and fix struct-null comparisons

### PR DESCRIPTION
**Pull request by Wraith2**
Strict is enabled and all struct-null comparisons are fixed up to check the value that is going to be passed to the log correctly in each case.

Fixes https://github.com/dotnet/corefx/issues/31476

### CODE REVIEW

File: src/System.Transactions.Local/src/System.Transactions.Local.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     There are many instances of comparing an instance of a struct with null. Until these are all fixed,
     disable the diagnostic.
     -->
-    <Features>$(Features.Replace('strict', '')</Features>
```
Comment (jkotas) [inline src/System.Transactions.Local/src/System.Transactions.Local.csproj]: Delete this instead (including the comment). This is enabled globally for corefx.

Comment (Wraith2) [reply src/System.Transactions.Local/src/System.Transactions.Local.csproj]: Done

**Comment by jkotas** [comment]
@jimcarley @qizhanMS @dmetzgar PTLA

**Comment by jimcarley** [comment]
@jkotas - LGTM

**Pull request merged**